akvo/iati/checks/fields/__init__.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+
+# Akvo RSR is covered by the GNU Affero General Public License.
+# See more details in the license.txt file located at the root folder of the Akvo RSR module.
+# For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+
+from .budgets import budgets
+from .conditions import conditions
+from .countries_and_regions import countries_and_regions
+from .country_budget_items import country_budget_items
+from .crs_add import crs_add
+from .dates import dates
+from .description import description
+from .document_links import document_links
+from .fss import fss
+from .humanitarian_scope import humanitarian_scope
+from .iati_identifier import iati_identifier
+from .legacy_data import legacy_data
+from .locations import locations
+from .partners import partners
+from .planned_disbursements import planned_disbursements
+from .policy_markers import policy_markers
+from .related_activities import related_activities
+from .reporting_org import reporting_org
+from .results import results
+from .sectors import sectors
+from .status import status
+from .title import title
+from .transactions import transactions
+
+
+__all__ = [
+    'budgets',
+    'conditions',
+    'countries_and_regions',
+    'country_budget_items',
+    'crs_add',
+    'dates',
+    'description',
+    'document_links',
+    'fss',
+    'humanitarian_scope',
+    'iati_identifier',
+    'legacy_data',
+    'locations',
+    'partners',
+    'planned_disbursements',
+    'policy_markers',
+    'related_activities',
+    'reporting_org',
+    'results',
+    'sectors',
+    'status',
+    'title',
+    'transactions',
+]
```

Comment:
I think there's enough code duplication in the fields functions to warrant some abstraction for the field checks module. The most common check seems to be if a field is empty or not. And if it is the action is the same; set **all_checks_passed** to False and add an error string.

Sketch of a base class and an implementing class:

```python
class FieldBase(object):
    def __init__(self, project):
        self.project = project
        self.all_checks_passed = True
        self.errors = []
        self.warnings = []

    def required(self, obj, fields):
        for field in fields:
            if not getattr(obj, field, False):
                self.all_checks_passed = False
                self.errors.append(u'{} (id: {}) has no {}'.format(
                        self.__class__.__name__.lower(), obj.pk, field.replace('_', ' ')))

    def execute(self):
        for model in models:
            model.check()
            self.all_checks_passed = model.all_checks_passed or self.all_checks_passed
            self.errors += model.errors
            self.warnings += model.warnings


class Budget(FieldBase):
    def check(self):
        required_fields = ['amount', 'period_start', 'period_end', 'value_date']
        for budget in self.project.budget_items.all():
            self.required(budget, required_fields)

            if not budget.currency and not self.project.currency:
                self.all_checks_passed = False
                self.errors.append(u'budget (id: 'u'{}) has no currency and no default currency '
                                   u'specified'.format(budget.pk))
```